controllers/gitrepository_controller.go

```diff
@@ -755,15 +728,34 @@ func (r *GitRepositoryReconciler) gitCheckout(ctx context.Context,
 			Err:    fmt.Errorf("failed to configure checkout strategy for Git implementation '%s': %w", obj.Spec.GitImplementation, err),
 			Reason: sourcev1.GitOperationFailedReason,
 		}
-		conditions.MarkTrue(obj, sourcev1.FetchFailedCondition, e.Reason, e.Err.Error())
```

Comment:
Why has this been removed?

Reply:
This was removed in a previous commit where I was checking the error is stalling in `reconcileSource` and setting the condition there. After #736 was merged this should've been reverted.